contracts/oracles/balancer/BPTWeightedPriceFeed.sol

```diff
@@ -193,11 +199,11 @@ contract BPTWeightedPriceFeed is LPPriceFeed {
             }
         }
 
-        answer = int256(weightedPrice / WAD_OVER_USD_FEED_SCALE); // U:[BAL-W-2]
+        answer = int256(weightedPrice / (numAssets * WAD_OVER_USD_FEED_SCALE)); // U:[BAL-W-2]
     }
 
     function getLPExchangeRate() public view override returns (uint256) {
-        return _getBPTInvariant().divDown(_getBPTSupply()); // U:[BAL-W-1]
+        return (numAssets * _getBPTInvariant()).divDown(_getBPTSupply()); // U:[BAL-W-1]
```

Comment:
Note that the resulting price, which is a product of exchange rate and aggregate price, doesn't change, except for maybe small rounding errors.

This way, however, numbers are more interpretable, and lower bound can be set based on pool's `getRate`.